### PR TITLE
Image building and unit tests for cluster-api-provider-aws

### PR DESCRIPTION
**Pull request by paulfantom**
Added jobs to build images and run unit tests with ci-operator.

/hold

### CODE REVIEW

**Comment by bbguimaraes** [comment]
@paulfantom: commands have to be on a separate line.

/hold

**Comment by bbguimaraes** [comment]
The command won't be recognized if there's text after it on the same line:

/hold cancel *text after it on the same line*

File: ci-operator/jobs/openshift/cluster-api-provider-aws/openshift-cluster-api-provider-aws-presubmits.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+presubmits:
+  openshift/cluster-api-provider-aws:
```
Comment (stevekuznetsov) [inline ci-operator/jobs/openshift/cluster-api-provider-aws/openshift-cluster-api-provider-aws-presubmits.yaml]: Likely want a `--target=[images]` presubmit as well

File: cluster/ci/config/prow/plugins.yaml [hunk 2]
```diff
@@ -433,3 +435,8 @@ plugins:
   - trigger
   - approve
   - blunderbuss
+
+  openshift/cluster-api-provider-aws:
```
Comment (stevekuznetsov) [inline cluster/ci/config/prow/plugins.yaml]: After #1229 you don't need these

Comment (paulfantom) [reply cluster/ci/config/prow/plugins.yaml]: Even `approve`?

File: ci-operator/jobs/openshift/cluster-api-provider-aws/openshift-cluster-api-provider-aws-presubmits.yaml (hunk 1 above)
Comment (paulfantom) [reply ci-operator/jobs/openshift/cluster-api-provider-aws/openshift-cluster-api-provider-aws-presubmits.yaml]: You're right. Added. :+1:

Comment (stevekuznetsov) [reply ci-operator/jobs/openshift/cluster-api-provider-aws/openshift-cluster-api-provider-aws-presubmits.yaml]: I would suggest separate jobs

**Comment by petr-muller** [comment]
We'll split the jobs when we fire up the generator anyway.

**Pull request merged**